Galaxy app integrations

phulsechinmay opened the pull request:
Closes #45 

**Features:**
* Added a `GalaxyIntegrationConfig` interface that specifies the UI / messaging overrides that a Galaxy app can define
* Added a function that returns a config with default messaging for overrides that don't want to be explicitly defined
* Created a decorator that returns the appropriate integration config according to the redirectPath specified in the request query params
  * Uses longest matching prefix to find the config to use

phulsechinmay commented:
@geooot Just needed to get some ideation on this. Currently all the Galaxy app integrations are defined under a constants file. If we're going to open this up so messaging is stored in some database and can be edited by hand / through some defined endpoints, then this PR will quite a few changes.

However, if we're good with them being defined in a constants file right now, I've also added a decorator that returns the config according to the redirect path.

Something to consider would be that Gatekeeper would need to be redeployed everytime we change this constants file.

geooot commented:
@phulsechinmay I think the strings should just be in a constants file, no need to store it in a database

geooot commented on src/galaxy-integrations/constants/default-integrations.ts:
```diff
@@ -0,0 +1,18 @@
+import { newGalaxyIntegrationConfig } from "../interfaces/galaxy-intergration";
```
Make spelling of `galaxy-intergtation` correct

geooot commented on src/galaxy-integrations/interfaces/galaxy-intergration.ts:
```diff
@@ -0,0 +1,35 @@
+  signupPageSubtext?: string;
+}
+
+export const newGalaxyIntegrationConfig = (
```
Add a JSDoc for this

geooot commented on src/galaxy-integrations/decorators/redirect-galaxy-integration.decorator.ts:
```diff
@@ -0,0 +1,37 @@
+  GatekeeperGalaxyIntegration
+} from "../constants/default-integrations";
+
+const getLongestMatchingPrefix = (key, prefixes) => {
```
Add a JSDoc

phulsechinmay commented:
@geooot I added the test for the `getLongestPathPrefix` function and fixed it to be more logically correct. About the test for the decorator, seems like it's a pain to test decorators in NestJS ([source](https://medium.com/@enrico.ferro/create-a-test-case-for-custom-parameter-decorator-in-nestjs-7-d97de5c4e6a0)). 

Looking at that, do you think a test for it is worth it?
